backends/cortex_m/passes/convert_to_cortex_m_pass.py

```diff
@@ -156,11 +156,24 @@ def _get_convolution_replacement(self, node) -> int:
             quantized_multipliers.append(quantized_multiplier)
             quantized_shifts.append(quantized_shift)
 
-        # Permute the weight tensor to the OHWI layout expected by CMSIS-NN.
         weight_tensor = get_param_tensor(self.exported_program, weight)
-        weight_permuted = weight_tensor.permute(0, 2, 3, 1).contiguous(
-            memory_format=torch.channels_last
-        )
+
+        # Detect depthwise convolution:
+        # PyTorch depthwise weight is [out_ch, 1, H, W] where dimension 1 is 1
+        # and groups == input_channels (groups > 1)
+        is_depthwise = weight_tensor.shape[1] == 1 and groups > 1
```

Comment:
I think groups could be 1 for a DW conv?
So a better condition is then, is_depthwise = (in_channels == groups) and (out_channels % in_channels) == 0

Reply:
Done.